### ISSUE TITLE
Unit tests fails

### ISSUE THREAD

**Issue by Apteryks** [supercollider/supercollider#3555]
Hello,

When running the unit tests of SuperCollider with Guix (I'm packaging SuperCollider for it right now), they fail like:

```
 Running tests...
/gnu/store/6g4pfzgmgwpfkwix0p7r6j349hffh2gs-cmake-3.7.2/bin/ctest --force-new-ctest-process 
Test project /tmp/guix-build-supercollider-3.9.1.drv-0/build
      Start  1: buffer_manager_test_run
 1/19 Test  #1: buffer_manager_test_run ...........................................................................................   Passed    0.03 sec
      Start  2: callback_interpreter_system_run
 2/19 Test  #2: callback_interpreter_system_run ...................................................................................   Passed    0.02 sec
      Start  3: memory-pool_test_run
 3/19 Test  #3: memory-pool_test_run ..............................................................................................   Passed    0.02 sec
      Start  4: server_dsp_thread_queue_test_run
 4/19 Test  #4: server_dsp_thread_queue_test_run ..................................................................................   Passed    1.15 sec
      Start  5: server_dsp_thread_test_run
 5/19 Test  #5: server_dsp_thread_test_run ........................................................................................   Passed    4.78 sec
      Start  6: server_node_graph_test_run
 6/19 Test  #6: server_node_graph_test_run ........................................................................................   Passed    0.02 sec
      Start  7: server_scheduler_test_run
 7/19 Test  #7: server_scheduler_test_run .........................................................................................   Passed    0.47 sec
      Start  8: server_synth_factory_test_run
 8/19 Test  #8: server_synth_factory_test_run .....................................................................................   Passed    0.02 sec
      Start  9: server_test_run
 9/19 Test  #9: server_test_run ...................................................................................................***Exception: SegFault  0.29 sec
SC FFT global init: cosTable initialised.
Running 3 test cases...
unknown location(0): fatal error: in "server_test_1": memory access violation at address: 0x00000000: no mapping at fault address
/tmp/guix-build-supercollider-3.9.1.drv-0/SuperCollider-Source/testsuite/supernova/server_test.cpp(29): last checkpoint: "server_test_1" entry.

*** 1 failure is detected in the test module "Master Test Suite"

      Start 10: simple_pool_test_run
10/19 Test #10: simple_pool_test_run ..............................................................................................   Passed    0.02 sec
      Start 11: sized_array_test_run
11/19 Test #11: sized_array_test_run ..............................................................................................   Passed    0.02 sec
      Start 12: sndfile_backend_test_run
12/19 Test #12: sndfile_backend_test_run ..........................................................................................   Passed    1.02 sec
      Start 13: static_pool_test_run
13/19 Test #13: static_pool_test_run ..............................................................................................   Passed    0.02 sec
      Start 14: timetag_test_run
14/19 Test #14: timetag_test_run ..................................................................................................   Passed    0.02 sec
      Start 15: perf_counter_test_run
15/19 Test #15: perf_counter_test_run .............................................................................................***Failed    0.02 sec
Running 1 test case...
unknown location(0): fatal error: in "perf_count_test": std::exception: std::exception
/tmp/guix-build-supercollider-3.9.1.drv-0/SuperCollider-Source/testsuite/supernova/perf_counter_test.cpp(6): last checkpoint: "perf_count_test" entry.

*** 1 failure is detected in the test module "Master Test Suite"

      Start 16: /tmp/guix-build-supercollider-3.9.1.drv-0/SuperCollider-Source/testsuite/sclang/sclang_crash_1.scd_run
16/19 Test #16: /tmp/guix-build-supercollider-3.9.1.drv-0/SuperCollider-Source/testsuite/sclang/sclang_crash_1.scd_run ............***Exception: Other  0.35 sec
/tmp/guix-build-supercollider-3.9.1.drv-0/build/lang/sclang: Relink `/gnu/store/b9ww6qv1ii9v6n45kin7543vkf6jfnd3-libpng-1.6.29/lib/libpng16.so.16' with `/gnu/store/n6nvxlk2j8ysffjh3jphn1k5silnakh6-glibc-2.25/lib/libpthread.so.0' for IFUNC symbol `longjmp'
/tmp/guix-build-supercollider-3.9.1.drv-0/build/lang/sclang: Relink `/gnu/store/1jp44pfmqcj3zycclvmmva3xcwinyg7l-freetype-2.8/lib/libfreetype.so.6' with `/gnu/store/n6nvxlk2j8ysffjh3jphn1k5silnakh6-glibc-2.25/lib/libpthread.so.0' for IFUNC symbol `longjmp'
SC FFT global init: cosTable initialised.
Exception while parsing YAML config file: operator[] call on a scalar
Exception while parsing YAML config file: operator[] call on a scalar
compiling class library...
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/guix-build-supercollider-3.9.1.drv-0/runtime-nixbld'
qt.qpa.screen: QXcbConnection: Could not connect to display 
Could not connect to any X display.
terminate called without an active exception

      Start 17: /tmp/guix-build-supercollider-3.9.1.drv-0/SuperCollider-Source/testsuite/sclang/sclang_crash_2.scd_run
17/19 Test #17: /tmp/guix-build-supercollider-3.9.1.drv-0/SuperCollider-Source/testsuite/sclang/sclang_crash_2.scd_run ............***Exception: Other  0.20 sec
/tmp/guix-build-supercollider-3.9.1.drv-0/build/lang/sclang: Relink `/gnu/store/b9ww6qv1ii9v6n45kin7543vkf6jfnd3-libpng-1.6.29/lib/libpng16.so.16' with `/gnu/store/n6nvxlk2j8ysffjh3jphn1k5silnakh6-glibc-2.25/lib/libpthread.so.0' for IFUNC symbol `longjmp'
/tmp/guix-build-supercollider-3.9.1.drv-0/build/lang/sclang: Relink `/gnu/store/1jp44pfmqcj3zycclvmmva3xcwinyg7l-freetype-2.8/lib/libfreetype.so.6' with `/gnu/store/n6nvxlk2j8ysffjh3jphn1k5silnakh6-glibc-2.25/lib/libpthread.so.0' for IFUNC symbol `longjmp'
SC FFT global init: cosTable initialised.
Exception while parsing YAML config file: operator[] call on a scalar
Exception while parsing YAML config file: operator[] call on a scalar
compiling class library...
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/guix-build-supercollider-3.9.1.drv-0/runtime-nixbld'
qt.qpa.screen: QXcbConnection: Could not connect to display 
Could not connect to any X display.
terminate called without an active exception

      Start 18: /tmp/guix-build-supercollider-3.9.1.drv-0/SuperCollider-Source/testsuite/sclang/sclang_crash_3.scd_run
18/19 Test #18: /tmp/guix-build-supercollider-3.9.1.drv-0/SuperCollider-Source/testsuite/sclang/sclang_crash_3.scd_run ............***Exception: Other  0.20 sec
/tmp/guix-build-supercollider-3.9.1.drv-0/build/lang/sclang: Relink `/gnu/store/b9ww6qv1ii9v6n45kin7543vkf6jfnd3-libpng-1.6.29/lib/libpng16.so.16' with `/gnu/store/n6nvxlk2j8ysffjh3jphn1k5silnakh6-glibc-2.25/lib/libpthread.so.0' for IFUNC symbol `longjmp'
/tmp/guix-build-supercollider-3.9.1.drv-0/build/lang/sclang: Relink `/gnu/store/1jp44pfmqcj3zycclvmmva3xcwinyg7l-freetype-2.8/lib/libfreetype.so.6' with `/gnu/store/n6nvxlk2j8ysffjh3jphn1k5silnakh6-glibc-2.25/lib/libpthread.so.0' for IFUNC symbol `longjmp'
SC FFT global init: cosTable initialised.
Exception while parsing YAML config file: operator[] call on a scalar
Exception while parsing YAML config file: operator[] call on a scalar
compiling class library...
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/guix-build-supercollider-3.9.1.drv-0/runtime-nixbld'
qt.qpa.screen: QXcbConnection: Could not connect to display 
Could not connect to any X display.
terminate called without an active exception

      Start 19: /tmp/guix-build-supercollider-3.9.1.drv-0/SuperCollider-Source/testsuite/sclang/sclang_crash_deprecated.scd_run
19/19 Test #19: /tmp/guix-build-supercollider-3.9.1.drv-0/SuperCollider-Source/testsuite/sclang/sclang_crash_deprecated.scd_run ...***Exception: Other  0.20 sec
/tmp/guix-build-supercollider-3.9.1.drv-0/build/lang/sclang: Relink `/gnu/store/b9ww6qv1ii9v6n45kin7543vkf6jfnd3-libpng-1.6.29/lib/libpng16.so.16' with `/gnu/store/n6nvxlk2j8ysffjh3jphn1k5silnakh6-glibc-2.25/lib/libpthread.so.0' for IFUNC symbol `longjmp'
/tmp/guix-build-supercollider-3.9.1.drv-0/build/lang/sclang: Relink `/gnu/store/1jp44pfmqcj3zycclvmmva3xcwinyg7l-freetype-2.8/lib/libfreetype.so.6' with `/gnu/store/n6nvxlk2j8ysffjh3jphn1k5silnakh6-glibc-2.25/lib/libpthread.so.0' for IFUNC symbol `longjmp'
SC FFT global init: cosTable initialised.
Exception while parsing YAML config file: operator[] call on a scalar
Exception while parsing YAML config file: operator[] call on a scalar
compiling class library...
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/guix-build-supercollider-3.9.1.drv-0/runtime-nixbld'
qt.qpa.screen: QXcbConnection: Could not connect to display 
Could not connect to any X display.
terminate called without an active exception


68% tests passed, 6 tests failed out of 19

Total Test time (real) =   8.92 sec

The following tests FAILED:
          9 - server_test_run (SEGFAULT)
         15 - perf_counter_test_run (Failed)
         16 - /tmp/guix-build-supercollider-3.9.1.drv-0/SuperCollider-Source/testsuite/sclang/sclang_crash_1.scd_run (OTHER_FAULT)
         17 - /tmp/guix-build-supercollider-3.9.1.drv-0/SuperCollider-Source/testsuite/sclang/sclang_crash_2.scd_run (OTHER_FAULT)
         18 - /tmp/guix-build-supercollider-3.9.1.drv-0/SuperCollider-Source/testsuite/sclang/sclang_crash_3.scd_run (OTHER_FAULT)
         19 - /tmp/guix-build-supercollider-3.9.1.drv-0/SuperCollider-Source/testsuite/sclang/sclang_crash_deprecated.scd_run (OTHER_FAULT)
Errors while running CTest
```

Any ideas? Is this to be expected when running the tests in a containerized environment?

**Comment by brianlheim** [comment]
The supernova test failures are known (#1717). I'm not actually sure if/why the files in testsuite/sclang get turned into test executables. Either way, we currently don't run these on CI.

**Comment by Apteryks** [comment]
Hello Brian, and thank you for the prompt reply! I'll disable the tests for now. Feel free to merge those issues if deemed appropriate. Thanks!

**Comment by brianlheim** [comment]
@Apteryks - no problem. Thanks for bringing this to our attention; I wasn't aware of those tests in testsuite/sclang. If you want, check out #3556 which fixes the issues with those four targets. It won't resolve the supernova test failures, though, so probably best to simply disable things for now.

And, thanks for working on packaging for guix!

**Comment by telephon** [comment]
yes!


it would be so good to have a system like guix for the sclang quarks

**Comment by Apteryks** [comment]
@telephon SuperCollider should be on its way to be merged into Guix proper: http://debbugs.gnu.org/cgi/bugreport.cgi?bug=30851. Feel free to elaborate from there :).

**Comment by dvzrv** [comment]
As I'm basically in the same boat as @Apteryks, is there any way of deselecting the failing tests?
Probably faking X11 with XVFB? @Apteryks have you tried that or disabled all of the tests?

```
Running tests...
UpdateCTestConfiguration  from :/build/supercollider/src/supercollider-3.10.0/build/DartConfiguration.tcl        
Parse Config file:/build/supercollider/src/supercollider-3.10.0/build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/build/supercollider/src/supercollider-3.10.0/build/DartConfiguration.tcl
Parse Config file:/build/supercollider/src/supercollider-3.10.0/build/DartConfiguration.tcl
Test project /build/supercollider/src/supercollider-3.10.0/build                                                            
Constructing a list of tests         
Done constructing a list of tests                                        
Updating test list for fixtures                                                         
Added 0 tests to meet fixture requirements                                                                                                   
Checking test dependency graph...                                                       
Checking test dependency graph end                                                                                                           
test 1                                                                                  
      Start  1: scsynth_test_run                                                                                                             
   
1: Test command: /build/supercollider/src/supercollider-3.10.0/build/testsuite/server/scsynth/scsynth_test            
1: Test timeout computed to be: 1500
1: Running 1 test case...                                                
1:     
1: *** No errors detected            
1: 
 1/20 Test  #1: scsynth_test_run ...................   Passed    0.10 sec                                         
test 2                               
      Start  2: buffer_manager_test_run   
                                                                                          
2: Test command: /build/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/buffer_manager_test                                 
2: Test timeout computed to be: 1500
2: Running 1 test case...                                                
2:     
2: *** No errors detected                                                
2:     
 2/20 Test  #2: buffer_manager_test_run ............   Passed    0.00 sec                                        
test 3                               
      Start  3: callback_interpreter_system_run                                                                                                                                                                                                                                                                                                                                               
                                     
3: Test command: /build/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/callback_interpreter_system
3: Test timeout computed to be: 1500                           
3: Running 3 test cases...                                               
3:                                              
3: *** No errors detected                                                                
3:     
 3/20 Test  #3: callback_interpreter_system_run ....   Passed    0.00 sec                                            
test 4                               
      Start  4: memory-pool_test_run                                                                                                                                                                                                                                                                                                                                                          
                                     
4: Test command: /build/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/memory-pool_test         
4: Test timeout computed to be: 1500                           
4: Running 3 test cases...                                               
4:                                              
4: *** No errors detected                                                                
4:     
 4/20 Test  #4: memory-pool_test_run ...............   Passed    0.00 sec                                        
test 5                               
      Start  5: server_dsp_thread_queue_test_run                                                                                                                                                                                                                                                                                                                                              
                                     
5: Test command: /build/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/server_dsp_thread_queue_test
5: Test timeout computed to be: 1500                           
5: Running 5 test cases...                                               
5:                                                                                      
5: *** No errors detected                                                                                                                    
5:                                                                                      
 5/20 Test  #5: server_dsp_thread_queue_test_run ...   Passed    0.52 sec                                                                    
test 6                                                                                  
      Start  6: server_dsp_thread_test_run                                                                                                                                                                                                                                                                                                                                                    
                                     
6: Test command: /build/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/server_dsp_thread_test
6: Test timeout computed to be: 1500                           
6: Running 6 test cases...                                               
6:                                              
6: *** No errors detected                                                                
6: 
 6/20 Test  #6: server_dsp_thread_test_run .........   Passed    2.11 sec                                         
test 7                                                          
      Start  7: server_node_graph_test_run
                                                                                          
7: Test command: /build/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/server_node_graph_test                              
7: Test timeout computed to be: 1500  
7: Running 19 test cases...                                              
7:                                              
7: *** No errors detected                                                
7:                                              
 7/20 Test  #7: server_node_graph_test_run .........   Passed    0.01 sec                                        
test 8                               
      Start  8: server_scheduler_test_run                                                                                                                                                                                                             
                                                                
8: Test command: /build/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/server_scheduler_test                                                                                        
8: Test timeout computed to be: 1500                                                                                                                                                                                                                                                                                                                                                         
8: Running 2 test cases...                                                                                                                                                                                   
8:                                                                                                                                                                                                                                                                                                                                                                                           
8: *** No errors detected                                                                                               
8:                                                                                                                      
 8/20 Test  #8: server_scheduler_test_run ..........   Passed    0.21 sec                                                                                                                                                                                                                                                                                                                     
test 9                                                                                                                                                                                                       
      Start  9: server_synth_factory_test_run                                                                                                                                                                                                         
                                                                                                                  
9: Test command: /build/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/server_synth_factory_test                                                                                       
9: Test timeout computed to be: 1500                                                                                                                                                                            
9: Running 1 test case...                                                                                                       
9:                                                                                                                                                                                                              
9: *** No errors detected                                                                                                      
9:                                                                                                                      
 9/20 Test  #9: server_synth_factory_test_run ......   Passed    0.00 sec                                                                                                                                                                                                                                                                                                                     
test 10                                                                                                                             
      Start 10: server_test_run                                                                                                                                                                                                                       
                                                                                                                                                                                                             
10: Test command: /build/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/server_test                
10: Test timeout computed to be: 1500                                                                                                                                                                           
10: Running 3 test cases...                                                                                                                                                                                  
10: unknown location(0): fatal error: in "server_test_1": std::bad_alloc: std::bad_alloc                                       
10: /build/supercollider/src/supercollider-3.10.0/testsuite/server/supernova/server_test.cpp(29): last checkpoint: "server_test_1" test entry
10: unknown location(0): fatal error: in "server_test_2": std::bad_alloc: std::bad_alloc                                                                                                                        
10: /build/supercollider/src/supercollider-3.10.0/testsuite/server/supernova/server_test.cpp(48): last checkpoint: "server_test_2" test entry                                                                                                                                                                                                                                                 
10: unknown location(0): fatal error: in "server_test_3": std::bad_alloc: std::bad_alloc                                                                                                                     
10: /build/supercollider/src/supercollider-3.10.0/testsuite/server/supernova/server_test.cpp(66): last checkpoint: "server_test_3" test entry                                                                                                                  
10:                                                                                                                          
10: *** 3 failures are detected in the test module "Master Test Suite"                                                                                                                                       
10:                                                                                                                                                                                                          
10/20 Test #10: server_test_run ....................***Failed    0.11 sec                                                      
test 11                                                                                                                                                                                                         
      Start 11: simple_pool_test_run                                                                                                                                                                         
                                                                                                                    
11: Test command: /build/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/simple_pool_test                                                                                               
11: Test timeout computed to be: 1500                                                                                                                                                                           
11: Running 1 test case...                                                                                                                          
11:                                                                                                                                                                                                             
11: *** No errors detected                                                                                                                                                                                      
11:                                                                                                                       
11/20 Test #11: simple_pool_test_run ...............   Passed    0.00 sec                                                                                                                                       
test 12                                                                                                                     
      Start 12: sized_array_test_run                                                                                
                                                                                                                                                                                                                
12: Test command: /build/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/sized_array_test                                                                                                                                     
12: Test timeout computed to be: 1500                                                                                                                                                                        
12: Running 4 test cases...                                                                                                                                                                                                                           
12:                                                                                                         
12: *** No errors detected                                                                                                                                                                                                                                                                                                                                                           [100/9518]
12:                                                                                                                                                                                                          
12/20 Test #12: sized_array_test_run ...............   Passed    0.01 sec                                                                                                                                    
test 13                                                                                                                                                                                                                                                                                                                                                                                       
      Start 13: sndfile_backend_test_run                                                                                                                                                                        
                                                                                                                        
13: Test command: /build/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/sndfile_backend_test                                                                                                                                 
13: Test timeout computed to be: 1500                                                                                                                                                                           
13: Running 1 test case...                                                                                                                                                                                                                            
13:                                                                                                               
13: *** No errors detected                                                                                                                                                                                      
13:                                                                                                                                                                                                             
13/20 Test #13: sndfile_backend_test_run ...........   Passed    1.00 sec                                                       
test 14                                                                                                                                                                                                         
      Start 14: static_pool_test_run                                                                                           
                                                                                                                        
14: Test command: /build/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/static_pool_test                                                                                                                                     
14: Test timeout computed to be: 1500                                                                                               
14: Running 2 test cases...                                                                                                                                                                                                                           
14:                                                                                                                                                                                                          
14: *** No errors detected                                                                                                  
14:                                                                                                                                                                                                             
14/20 Test #14: static_pool_test_run ...............   Passed    0.00 sec                                                                                                                                    
test 15                                                                                                                        
      Start 15: timetag_test_run                                                                                                             
                                                                                                                                                                                                                
15: Test command: /build/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/timetag_test                                                                                                                                                  
15: Test timeout computed to be: 1500                                                                                                                                                                        
15: Running 3 test cases...                                                                                                                                                                                                                                    
15:                                                                                                                          
15: *** No errors detected                                                                                                                                                                                   
15:                                                                                                                                                                                                             
15/20 Test #15: timetag_test_run ...................   Passed    0.00 sec                                                      
test 16                                                                                                                                                                                                         
      Start 16: perf_counter_test_run                                                                                                                                                                        
                                                                                                                    
16: Test command: /build/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/perf_counter_test                                                                                              
16: Test timeout computed to be: 1500                                                                                                                                                                           
16: Running 1 test case...                                                                                                                          
16: unknown location(0): fatal error: in "perf_count_test": std::exception: std::exception                                                                                                                      
16: /build/supercollider/src/supercollider-3.10.0/testsuite/server/supernova/perf_counter_test.cpp(6): last checkpoint: "perf_count_test" test entry                                                            
16:                                                                                                                       
16: *** 1 failure is detected in the test module "Master Test Suite"                                                                                                                                            
16:                                                                                                                         
16/20 Test #16: perf_counter_test_run ..............***Failed    0.00 sec                                           
test 17                                                                                                                                                                                                         
      Start 17: sclang_crash_1_run                                                                                                                                                                                                                                                                                                                                                            
                                                                                                                                                                                                             
17: Test command: /build/supercollider/src/supercollider-3.10.0/build/lang/sclang "-a" "-l" "/build/supercollider/src/supercollider-3.10.0/build/build_sclang.cfg" "/build/supercollider/src/supercollider-3.10.0/testsuite/sclang/sclang_crash_1.scd"
17: Test timeout computed to be: 1500                                                                                                                                                                        
17: compiling class library...                                                                                                                                                                               
17: qt.qpa.screen: QXcbConnection: Could not connect to display                                                                                                                                                                                                                                                                                                                              
17: Could not connect to any X display.                                                                                                                                                                      
17: terminate called without an active exception                                                                                                                                                                                                                                                                                                                                             
17/20 Test #17: sclang_crash_1_run .................***Exception: Child aborted  0.45 sec                               
test 18                                                                                                                 
      Start 18: sclang_crash_2_run                                                                                                                                                                                                                                                                                                                                                            
                                                                                                                                                                                                             
18: Test command: /build/supercollider/src/supercollider-3.10.0/build/lang/sclang "-a" "-l" "/build/supercollider/src/supercollider-3.10.0/build/build_sclang.cfg" "/build/supercollider/src/supercollider-3.10.0/testsuite/sclang/sclang_crash_2.scd"
18: Test timeout computed to be: 1500                                                                             
18: compiling class library...                                                                                                                                                                                  
18: qt.qpa.screen: QXcbConnection: Could not connect to display                                                                                                                                                 
18: Could not connect to any X display.                                                                                         
18: terminate called without an active exception                                                                                                                                                                
18/20 Test #18: sclang_crash_2_run .................***Exception: Child aborted  0.43 sec                                      
test 19                                                                                                                 
      Start 19: sclang_crash_3_run                                                                                                                                                                                                                                                                                                                                                            
                                                                                                                                    
19: Test command: /build/supercollider/src/supercollider-3.10.0/build/lang/sclang "-a" "-l" "/build/supercollider/src/supercollider-3.10.0/build/build_sclang.cfg" "/build/supercollider/src/supercollider-3.10.0/testsuite/sclang/sclang_crash_3.scd"
19: Test timeout computed to be: 1500                                                                                                                                                                        
19: compiling class library...                                                                                              
19: qt.qpa.screen: QXcbConnection: Could not connect to display                                                                                                                                                 
19: Could not connect to any X display.                                                                                                                                                                      
19: terminate called without an active exception                                                                               
19/20 Test #19: sclang_crash_3_run .................***Exception: Child aborted  0.40 sec                                                    
test 20                                                                                                                                                                                                         
      Start 20: sclang_crash_deprecated_run                                                                                                                                                                                                                                                                                                                                                   
                                                                                                                                                                                                             
20: Test command: /build/supercollider/src/supercollider-3.10.0/build/lang/sclang "-a" "-l" "/build/supercollider/src/supercollider-3.10.0/build/build_sclang.cfg" "/build/supercollider/src/supercollider-3.10.0/testsuite/sclang/sclang_crash_deprecated.scd"
20: Test timeout computed to be: 1500                                                                                        
20: compiling class library...                                                                                                                                                                               
20: qt.qpa.screen: QXcbConnection: Could not connect to display                                                                                                                                              
20: Could not connect to any X display.                                                                                        
20: terminate called without an active exception                                                                                                                                                                
20/20 Test #20: sclang_crash_deprecated_run ........***Exception: Child aborted  0.40 sec                                                                                                                    
                                                                                                                    
70% tests passed, 6 tests failed out of 20                                                                                                                                                                      
                                                                                                                                                                                                                
Total Test time (real) =   5.78 sec                                                                                                                 
                                                                                                                                                                                                                
The following tests FAILED:                                                                                                                                                                                     
         10 - server_test_run (Failed)                                                                                    
         16 - perf_counter_test_run (Failed)                                                                                                                                                                    
         17 - sclang_crash_1_run (Child aborted)                                                                            
         18 - sclang_crash_2_run (Child aborted)                                                                    
         19 - sclang_crash_3_run (Child aborted)                                                                                                                                                                
         20 - sclang_crash_deprecated_run (Child aborted)                                                                                                                                                                                             
Errors while running CTest                                                                                                       
```

When running in a X11 enabled environment only 10 and 16 fail, but this will not be the case for building the package on Arch Linux:

```
Running tests...
UpdateCTestConfiguration  from :/tmp/makepkg/supercollider/src/supercollider-3.10.0/build/DartConfiguration.tcl
Parse Config file:/tmp/makepkg/supercollider/src/supercollider-3.10.0/build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/tmp/makepkg/supercollider/src/supercollider-3.10.0/build/DartConfiguration.tcl
Parse Config file:/tmp/makepkg/supercollider/src/supercollider-3.10.0/build/DartConfiguration.tcl
Test project /tmp/makepkg/supercollider/src/supercollider-3.10.0/build
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 1
      Start  1: scsynth_test_run

1: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/testsuite/server/scsynth/scsynth_test
1: Test timeout computed to be: 1500
1: convolution: samplerate mismatch preset:48000 host:44100
1: convolution: samplerate mismatch preset:48000 host:44100
1: convolution: samplerate mismatch preset:48000 host:44100
1: Running 1 test case...
1: VST_PATH not set, defaulting to /home/dave/vst:/usr/local/lib/vst:/usr/lib/vst
1: convolution: samplerate mismatch preset:48000 host:44100
1: convolution: samplerate mismatch preset:48000 host:44100
1: convolution: samplerate mismatch preset:48000 host:44100
1: Found 442 LADSPA plugins
1: 
1: [1;32;49m*** No errors detected
1: [0;39;49m
 1/20 Test  #1: scsynth_test_run ...................   Passed    0.61 sec
test 2
      Start  2: buffer_manager_test_run

2: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/buffer_manager_test
2: Test timeout computed to be: 1500
2: Running 1 test case...
2: 
2: [1;32;49m*** No errors detected
2: [0;39;49m
 2/20 Test  #2: buffer_manager_test_run ............   Passed    0.00 sec
test 3
      Start  3: callback_interpreter_system_run

3: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/callback_interpreter_system
3: Test timeout computed to be: 1500
3: Running 3 test cases...
3: 
3: [1;32;49m*** No errors detected
3: [0;39;49m
 3/20 Test  #3: callback_interpreter_system_run ....   Passed    0.01 sec
test 4
      Start  4: memory-pool_test_run

4: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/memory-pool_test
4: Test timeout computed to be: 1500
4: Running 3 test cases...
4: 
4: [1;32;49m*** No errors detected
4: [0;39;49m
 4/20 Test  #4: memory-pool_test_run ...............   Passed    0.00 sec
test 5
      Start  5: server_dsp_thread_queue_test_run

5: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/server_dsp_thread_queue_test
5: Test timeout computed to be: 1500
5: Running 5 test cases...
5: 
5: [1;32;49m*** No errors detected
5: [0;39;49m
 5/20 Test  #5: server_dsp_thread_queue_test_run ...   Passed    0.52 sec
test 6
      Start  6: server_dsp_thread_test_run

6: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/server_dsp_thread_test
6: Test timeout computed to be: 1500
6: Running 6 test cases...
6: 
6: [1;32;49m*** No errors detected
6: [0;39;49m
 6/20 Test  #6: server_dsp_thread_test_run .........   Passed    2.09 sec
test 7
      Start  7: server_node_graph_test_run

7: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/server_node_graph_test
7: Test timeout computed to be: 1500
7: Running 19 test cases...
7: 
7: [1;32;49m*** No errors detected
7: [0;39;49m
 7/20 Test  #7: server_node_graph_test_run .........   Passed    0.01 sec
test 8
      Start  8: server_scheduler_test_run

8: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/server_scheduler_test
8: Test timeout computed to be: 1500
8: Running 2 test cases...
8: 
8: [1;32;49m*** No errors detected
8: [0;39;49m
 8/20 Test  #8: server_scheduler_test_run ..........   Passed    0.21 sec
test 9
      Start  9: server_synth_factory_test_run

9: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/server_synth_factory_test
9: Test timeout computed to be: 1500
9: Running 1 test case...
9: 
9: [1;32;49m*** No errors detected
9: [0;39;49m
 9/20 Test  #9: server_synth_factory_test_run ......   Passed    0.00 sec
test 10
      Start 10: server_test_run

10: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/server_test
10: Test timeout computed to be: 1500
10: Running 3 test cases...
10: unknown location(0): [4;31;49mfatal error: in "server_test_1": std::bad_alloc: std::bad_alloc[0;39;49m
10: /tmp/makepkg/supercollider/src/supercollider-3.10.0/testsuite/server/supernova/server_test.cpp(29): [1;36;49mlast checkpoint: "server_test_1" test entry[0;39;49m
10: unknown location(0): [4;31;49mfatal error: in "server_test_2": std::bad_alloc: std::bad_alloc[0;39;49m
10: /tmp/makepkg/supercollider/src/supercollider-3.10.0/testsuite/server/supernova/server_test.cpp(48): [1;36;49mlast checkpoint: "server_test_2" test entry[0;39;49m
10: unknown location(0): [4;31;49mfatal error: in "server_test_3": std::bad_alloc: std::bad_alloc[0;39;49m
10: /tmp/makepkg/supercollider/src/supercollider-3.10.0/testsuite/server/supernova/server_test.cpp(66): [1;36;49mlast checkpoint: "server_test_3" test entry[0;39;49m
10: 
10: [1;31;49m*** 3 failures are detected in the test module "Master Test Suite"
10: [0;39;49m
10/20 Test #10: server_test_run ....................***Failed    0.12 sec
test 11
      Start 11: simple_pool_test_run

11: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/simple_pool_test
11: Test timeout computed to be: 1500
11: Running 1 test case...
11: 
11: [1;32;49m*** No errors detected
11: [0;39;49m
11/20 Test #11: simple_pool_test_run ...............   Passed    0.00 sec
test 12
      Start 12: sized_array_test_run

12: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/sized_array_test
12: Test timeout computed to be: 1500
12: Running 4 test cases...
12: 
12: [1;32;49m*** No errors detected
12: [0;39;49m
12/20 Test #12: sized_array_test_run ...............   Passed    0.01 sec
test 13
      Start 13: sndfile_backend_test_run

13: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/sndfile_backend_test
13: Test timeout computed to be: 1500
13: Running 1 test case...
13: 
13: [1;32;49m*** No errors detected
13: [0;39;49m
13/20 Test #13: sndfile_backend_test_run ...........   Passed    1.00 sec
test 14
      Start 14: static_pool_test_run

14: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/static_pool_test
14: Test timeout computed to be: 1500
14: Running 2 test cases...
14: 
14: [1;32;49m*** No errors detected
14: [0;39;49m
14/20 Test #14: static_pool_test_run ...............   Passed    0.00 sec
test 15
      Start 15: timetag_test_run

15: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/timetag_test
15: Test timeout computed to be: 1500
15: Running 3 test cases...
15: 
15: [1;32;49m*** No errors detected
15: [0;39;49m
15/20 Test #15: timetag_test_run ...................   Passed    0.00 sec
test 16
      Start 16: perf_counter_test_run

16: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/testsuite/server/supernova/perf_counter_test
16: Test timeout computed to be: 1500
16: Running 1 test case...
16: unknown location(0): [4;31;49mfatal error: in "perf_count_test": std::exception: std::exception[0;39;49m
16: /tmp/makepkg/supercollider/src/supercollider-3.10.0/testsuite/server/supernova/perf_counter_test.cpp(6): [1;36;49mlast checkpoint: "perf_count_test" test entry[0;39;49m
16: 
16: [1;31;49m*** 1 failure is detected in the test module "Master Test Suite"
16: [0;39;49m
16/20 Test #16: perf_counter_test_run ..............***Failed    0.00 sec
test 17
      Start 17: sclang_crash_1_run

17: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/lang/sclang "-a" "-l" "/tmp/makepkg/supercollider/src/supercollider-3.10.0/build/build_sclang.cfg" "/tmp/makepkg/supercollider/src/supercollider-3.10.0/testsuite/sclang/sclang_crash_1.scd"
17: Test timeout computed to be: 1500
17: compiling class library...
17: 	Found 845 primitives.
17: 	Compiling directory '/tmp/makepkg/supercollider/src/supercollider-3.10.0/SCClassLibrary'
17: 	Compiling directory '/tmp/makepkg/supercollider/src/supercollider-3.10.0/platform/disable_startup_files'
17: 	numentries = 818594 / 11958602 = 0.068
17: 	5353 method selectors, 2234 classes
17: 	method table size 13112760 bytes, big table size 95668816
17: 	Number of Symbols 12128
17: 	Byte Code Size 364116
17: 	compiled 318 files in 0.41 seconds
17: 
17: Info: 5 methods are currently overwritten by extensions. To see which, execute:
17: MethodOverride.printAll
17: 
17: compile done
17: localhost : setting clientID to 0.
17: internal : setting clientID to 0.
17: Class tree inited in 0.01 seconds
17: WARNING: Extension in '/tmp/makepkg/supercollider/src/supercollider-3.10.0/platform/disable_startup_files/extPlatform.sc' overwrites Platform:loadStartupFiles in main class library.
17: Intentional overwrites must be put in a 'SystemOverwrites' subfolder.
17: 
17: 
17: *** Welcome to SuperCollider 3.10.0. *** For help type ctrl-c ctrl-h (Emacs) or :SChelp (vim) or ctrl-U (sced/gedit).
17: index: 0
17: nil
17: index: 1
17: nil
17: index: 2
17: Int8Array[ 100, 107, -14 ]
17: index: 3
17: nil
17: index: 4
17: nil
17: index: 5
17: nil
17: index: 6
17: nil
17: index: 7
17: nil
17: index: 8
17: nil
17: index: 9
17: { 1+1 }
17: cleaning up OSC
17/20 Test #17: sclang_crash_1_run .................   Passed    0.84 sec
test 18
      Start 18: sclang_crash_2_run

18: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/lang/sclang "-a" "-l" "/tmp/makepkg/supercollider/src/supercollider-3.10.0/build/build_sclang.cfg" "/tmp/makepkg/supercollider/src/supercollider-3.10.0/testsuite/sclang/sclang_crash_2.scd"
18: Test timeout computed to be: 1500
18: compiling class library...
18: 	Found 845 primitives.
18: 	Compiling directory '/tmp/makepkg/supercollider/src/supercollider-3.10.0/SCClassLibrary'
18: 	Compiling directory '/tmp/makepkg/supercollider/src/supercollider-3.10.0/platform/disable_startup_files'
18: 	numentries = 818594 / 11958602 = 0.068
18: 	5353 method selectors, 2234 classes
18: 	method table size 13112760 bytes, big table size 95668816
18: 	Number of Symbols 12128
18: 	Byte Code Size 364116
18: 	compiled 318 files in 0.41 seconds
18: 
18: Info: 5 methods are currently overwritten by extensions. To see which, execute:
18: MethodOverride.printAll
18: 
18: compile done
18: localhost : setting clientID to 0.
18: internal : setting clientID to 0.
18: Class tree inited in 0.01 seconds
18: WARNING: Extension in '/tmp/makepkg/supercollider/src/supercollider-3.10.0/platform/disable_startup_files/extPlatform.sc' overwrites Platform:loadStartupFiles in main class library.
18: Intentional overwrites must be put in a 'SystemOverwrites' subfolder.
18: 
18: 
18: *** Welcome to SuperCollider 3.10.0. *** For help type ctrl-c ctrl-h (Emacs) or :SChelp (vim) or ctrl-U (sced/gedit).
18: cleaning up OSC
18/20 Test #18: sclang_crash_2_run .................   Passed    0.82 sec
test 19
      Start 19: sclang_crash_3_run

19: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/lang/sclang "-a" "-l" "/tmp/makepkg/supercollider/src/supercollider-3.10.0/build/build_sclang.cfg" "/tmp/makepkg/supercollider/src/supercollider-3.10.0/testsuite/sclang/sclang_crash_3.scd"
19: Test timeout computed to be: 1500
19: compiling class library...
19: 	Found 845 primitives.
19: 	Compiling directory '/tmp/makepkg/supercollider/src/supercollider-3.10.0/SCClassLibrary'
19: 	Compiling directory '/tmp/makepkg/supercollider/src/supercollider-3.10.0/platform/disable_startup_files'
19: 	numentries = 818594 / 11958602 = 0.068
19: 	5353 method selectors, 2234 classes
19: 	method table size 13112760 bytes, big table size 95668816
19: 	Number of Symbols 12128
19: 	Byte Code Size 364116
19: 	compiled 318 files in 0.42 seconds
19: 
19: Info: 5 methods are currently overwritten by extensions. To see which, execute:
19: MethodOverride.printAll
19: 
19: compile done
19: localhost : setting clientID to 0.
19: internal : setting clientID to 0.
19: Class tree inited in 0.01 seconds
19: WARNING: Extension in '/tmp/makepkg/supercollider/src/supercollider-3.10.0/platform/disable_startup_files/extPlatform.sc' overwrites Platform:loadStartupFiles in main class library.
19: Intentional overwrites must be put in a 'SystemOverwrites' subfolder.
19: 
19: 
19: *** Welcome to SuperCollider 3.10.0. *** For help type ctrl-c ctrl-h (Emacs) or :SChelp (vim) or ctrl-U (sced/gedit).
19: cleaning up OSC
19/20 Test #19: sclang_crash_3_run .................   Passed    0.83 sec
test 20
      Start 20: sclang_crash_deprecated_run

20: Test command: /tmp/makepkg/supercollider/src/supercollider-3.10.0/build/lang/sclang "-a" "-l" "/tmp/makepkg/supercollider/src/supercollider-3.10.0/build/build_sclang.cfg" "/tmp/makepkg/supercollider/src/supercollider-3.10.0/testsuite/sclang/sclang_crash_deprecated.scd"
20: Test timeout computed to be: 1500
20: compiling class library...
20: 	Found 845 primitives.
20: 	Compiling directory '/tmp/makepkg/supercollider/src/supercollider-3.10.0/SCClassLibrary'
20: 	Compiling directory '/tmp/makepkg/supercollider/src/supercollider-3.10.0/platform/disable_startup_files'
20: 	numentries = 818594 / 11958602 = 0.068
20: 	5353 method selectors, 2234 classes
20: 	method table size 13112760 bytes, big table size 95668816
20: 	Number of Symbols 12128
20: 	Byte Code Size 364116
20: 	compiled 318 files in 0.39 seconds
20: 
20: Info: 5 methods are currently overwritten by extensions. To see which, execute:
20: MethodOverride.printAll
20: 
20: compile done
20: localhost : setting clientID to 0.
20: internal : setting clientID to 0.
20: Class tree inited in 0.01 seconds
20: WARNING: Extension in '/tmp/makepkg/supercollider/src/supercollider-3.10.0/platform/disable_startup_files/extPlatform.sc' overwrites Platform:loadStartupFiles in main class library.
20: Intentional overwrites must be put in a 'SystemOverwrites' subfolder.
20: 
20: 
20: *** Welcome to SuperCollider 3.10.0. *** For help type ctrl-c ctrl-h (Emacs) or :SChelp (vim) or ctrl-U (sced/gedit).
20: ERROR: throw during error handling!
20: 
20: Instance of DoesNotUnderstandError {    (0xaef7c1930b8, gc=54, fmt=00, flg=00, set=03)
20:   instance variables [6]
20:     what : Symbol 'DoesNotUnderstandError'
20:     protectedBacktrace : nil
20:     path : "/tmp/makepkg/supercollider/src/supercollider..."
20:     receiver : nil
20:     selector : Symbol 'ownerClass'
20:     args : instance of Array (0xaef791d9a80, size=0, set=0)
20: }
20: ERROR: throw during error handling!
20: 
20: Instance of DoesNotUnderstandError {    (0xaef7c1936e8, gc=54, fmt=00, flg=00, set=03)
20:   instance variables [6]
20:     what : Symbol 'DoesNotUnderstandError'
20:     protectedBacktrace : nil
20:     path : "/tmp/makepkg/supercollider/src/supercollider..."
20:     receiver : nil
20:     selector : Symbol 'name'
20:     args : instance of Array (0xaef791d9a80, size=0, set=0)
20: }
20: ERROR: throw during error handling!
20: 
20: Instance of DoesNotUnderstandError {    (0xaef79f1d128, gc=54, fmt=00, flg=00, set=03)
20:   instance variables [6]
20:     what : Symbol 'DoesNotUnderstandError'
20:     protectedBacktrace : nil
20:     path : "/tmp/makepkg/supercollider/src/supercollider..."
20:     receiver : nil
20:     selector : Symbol 'name'
20:     args : instance of Array (0xaef791d9a80, size=0, set=0)
20: }
20: ERROR: throw during error handling!
20: 
20: Instance of DoesNotUnderstandError {    (0xaef7c191958, gc=54, fmt=00, flg=00, set=03)
20:   instance variables [6]
20:     what : Symbol 'DoesNotUnderstandError'
20:     protectedBacktrace : nil
20:     path : "/tmp/makepkg/supercollider/src/supercollider..."
20:     receiver : nil
20:     selector : Symbol 'filenameSymbol'
20:     args : instance of Array (0xaef791d9a80, size=0, set=0)
20: }
20: WARNING: Called from {unknown}, method nil:nil is deprecated and will be removed.
20: The definition of 'nil' is to be found here: 'nil'
20: PATH: /tmp/makepkg/supercollider/src/supercollider-3.10.0/testsuite/sclang/sclang_crash_deprecated.scd
20: 
20: 
20: cleaning up OSC
20/20 Test #20: sclang_crash_deprecated_run ........   Passed    0.81 sec

90% tests passed, 2 tests failed out of 20

Total Test time (real) =   7.90 sec

The following tests FAILED:
	 10 - server_test_run (Failed)
	 16 - perf_counter_test_run (Failed)
Errors while running CTest
make: *** [Makefile:141: test] Error 8
```

**Comment by dvzrv** [comment]
@Apteryks thanks. I now run them in `xvfb-run` (for the tests depending on X11) and allow the remaining to fail by conditionally printing a warning during build.

However, 10 and 16 are still broken.
